ring/ring.go

```diff
@@ -590,3 +484,119 @@ func (r *Ring) EqualLvl(level int, p1, p2 *Poly) bool {
 
 	return true
 }
+
+// MarshalBinarySize returns the size in bytes of the target Ring.
+func (r *Ring) MarshalBinarySize() (dataLen int) {
```

Comment:
What is the use-case of full `ring.Ring` objects serialization ? It seems that they can easily be re-created from the degree and modulus slices with reasonable cost.